contracts/contracts/token/OUSD.sol

```diff
@@ -466,30 +466,33 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
      * @dev Ensures internal account for rebasing and non-rebasing credits and
      *      supply is updated following deployment of frozen yield change.
      */
-    function _ensureRebasingMigration(address _account) internal {
-        if (nonRebasingCreditsPerToken[_account] == 0) {
-            if (_creditBalances[_account] == 0) {
-                // Since there is no existing balance, we can directly set to
-                // high resolution, and do not have to do any other bookkeeping
-                nonRebasingCreditsPerToken[_account] = 1e27;
-            } else {
-                // Migrate the existing account:
-                // It is important that balanceOf not be called inside updating
-                // account data, since it will give wrong answers if it does
-                // not have all an account's data in a consistent state. This
-                // isn't a problem in the current implimentation, since we only
-                // need to update nonRebasingCreditsPerToken.
-                // Set fixed credits per token for this account
-                nonRebasingCreditsPerToken[_account] = _rebasingCreditsPerToken;
-
-                // Update global totals:
-                // Update non rebasing supply
-                nonRebasingSupply = nonRebasingSupply.add(balanceOf(_account));
-                // Update credit tallies
-                _rebasingCredits = _rebasingCredits.sub(
-                    _creditBalances[_account]
-                );
-            }
+    function _ensureMigrationToNonRebasing(address _account) internal {
+        if (nonRebasingCreditsPerToken[_account] != 0) {
+            return; // Account already is non-rebasing
+        }
+        if (_creditBalances[_account] == 0) {
+            // Since there is no existing balance, we can directly set to
+            // high resolution, and do not have to do any other bookkeeping
+            nonRebasingCreditsPerToken[_account] = 1e27;
+        } else {
+            // Get old values, so we can use them unaffected by changes
+            uint256 oldBalance = balanceOf(_account);
+            uint256 oldCredits = _creditBalances[_account];
+
+            // Atomicly update account information:
+            // It is important that balanceOf not be called inside updating
+            // account data, since it will give wrong answers if it does
+            // not have all an account's data in a consistent state.
+            nonRebasingCreditsPerToken[_account] = 1e27;
+            // difference between the 1e18 balance and the new 1e27 resolution
+            _creditBalances[_account] = oldBalance * 1e9;
```

Comment:
I like that we unified the non rebasing accounts to always having `nonRebasingCreditsPerToken` set to either 0 or 1e27. If we would at some point be able to transition all non rebasing accounts to this binary state the `nonRebasingCreditsPerToken` could become an account => bool map. 

Not just yet though because current nonRebasingAccounts have non binary value